packages/fast-usdc/src/exos/advancer.js

```diff
@@ -161,6 +158,26 @@ export const prepareAdvancerKit = (
             const destination = chainHub.makeChainAddress(EUD);
 
             const fullAmount = toAmount(evidence.tx.amount);
+            const {
+              tx: { forwardingAddress },
+              txHash,
+            } = evidence;
+
+            const { borrowerFacet, notifyFacet, poolAccount } = this.state;
+            if (
+              notifyFacet.forwardIfMinted(
+                destination,
+                forwardingAddress,
+                fullAmount,
+                txHash,
+              )
+            ) {
```

Comment:
surprising to see the condition be the result of an action. please make more clear.
```suggestion
            const forwarded = notifyFacet.forwardIfMinted(
                destination,
                forwardingAddress,
                fullAmount,
                txHash,
              );
            if (forwarded)) {
```

Reply:
This name was weird to me and as I get through more of the review, I understand better why. The `notify` facet shouldn't be telling the settler what to do.

What is it notifying it of? It's telling the settler about a transaction and it needs to know what decision the settler has to say about it. So I think this should be,
```suggestion
            const forwarded = notifyFacet.checkForwarded(
                destination,
                forwardingAddress,
                fullAmount,
                txHash,
              );
            if (forwarded) {
              return;
             }
```

I took out the logging too because the settler already does that. I'm not as confident in that change.

Reply:
This was tricky to name since "checkForwarded" does more than checking - it actually starts a Forward via MsgTransfer if conditions are met. That's why i wanted to include a verb.

I've heeded your suggestion, and renamed to `checkIfMinted`. (the forward hasn't occurred yet, and the early mint is the thing we're checking)

Reply:
> it actually starts a Forward via MsgTransfer if conditions are met. That's why i wanted to include a verb.

It happens to but that's needn't be a commitment to the caller. The caller is providing information and asking for a state.